flocker/control/_config.py

```diff
@@ -295,7 +297,8 @@ def convert_volume(self):
                 u'mountpoint': self._application.volume.mountpoint.path
             }
             dataset = self._application.volume.dataset
-            if dataset.dataset_id is not None:
+            if dataset.dataset_id != dataset_id_from_name(
+                    self._application.name):
```

Comment:
The reason it seems unnecessary to me is that if the dataset_id is included in the output then the next thing that looks at this data structure will just go ahead and use that id.  And if it isn't included then it will just re-generate the exact same value.  Either way, the result is the same.

But if we decide to change how the dataset_id is generated - for example, to support some kind of weak namespacing-like feature for `flocker up` (to let different developers use it simultaneously against one cluster) then it might get hard for the server to re-generate the same uuid.  Not that that means we should change this now but maybe it does mean we will eventually have to change it and fix all those tests.